packages/storefront/src/components/ComponentMetaOverview.vue

```diff
@@ -45,7 +45,94 @@
+        } else if (rowKey === 'eventNames') {
+          value = formatEvents(meta.eventsMeta);
+        } else if (rowKey === 'requiredProps') {
+          value = formatRequiredProps(meta.propsMeta);
+        }
+        const cellContent = formatCellContent(value, rowKey);
+        return `<p-table-cell>${cellContent}</p-table-cell>`;
+      };
+
+      const formatProps = (propsMeta: ComponentMeta['propsMeta']): string[] =>
+        Object.entries(propsMeta ?? {}).map(([propName, meta]) => {
+          const propFlags = getFlags(meta);
+          const formattedAllowedValues = formatAllowedValues(meta.allowedValues, !!meta.isDeprecated);
+          return (
+            `<span class="prop">${propName}${propFlags}</span>` +
+            (formattedAllowedValues ? `<div style="display: none;">${formattedAllowedValues}</div>` : '')
+          );
+        });
+
+      const getFlags = <
+        T extends { isDeprecated?: boolean; isBreakpointCustomizable?: boolean; isExperimental?: boolean },
+      >(
+        meta: T
+      ): string =>
+        [meta.isDeprecated && ' 🚫', meta.isBreakpointCustomizable && ' 🛠️', meta.isExperimental && ' 🧪️']
+          .filter(Boolean)
+          .join('');
+
+      const formatAllowedValues = (allowedValues: PropMeta['allowedValues'], isDeprecated: boolean): string => {
+        if (Array.isArray(allowedValues)) {
+          return allowedValues
+            .map((value) => (value === null ? 'undefined' : value))
+            .map((value) => (isDeprecated ? value + ' 🚫' : value))
+            .map((value) => `– ${value}`)
+            .join('<br>');
+        } else if (typeof allowedValues === 'object') {
+          return Object.entries(allowedValues)
+            .map(([key, val]) => {
+              if (Array.isArray(val)) {
+                val = val.map((v) => (v === null ? 'undefined' : v)).join(' | ');
+              }
+              return `- ${key}: ${val}`;
+            })
+            .join('<br>');
+        }
+        return `- ${allowedValues}`;
+      };
+
+      const formatEvents = (eventsMeta: ComponentMeta['eventsMeta']): string[] =>
+        Object.entries(eventsMeta ?? {}).map(([eventName, value]) => `${eventName}${getFlags(value)}`);
+
+      const formatRequiredProps = (propsMeta: ComponentMeta['propsMeta']): string =>
+        Object.entries(propsMeta ?? {})
+          .filter(([, { isRequired }]) => isRequired)
+          .map(([propName]) => `<code>${propName}</code>`)
+          .join('<br>');
+
+      const formatCellContent = (
+        value: boolean | string | string[] | TagName[] | undefined,
+        rowKey: string
+      ): string => {
+        if (!value) return '';
+        if (value === true) return '✅';
+        if (Array.isArray(value)) {
+          return value
+            .sort()
+            .map((val) => (rowKey === 'nestedComponents' ? val : `<code>${val}</code>`))
+            .join('<br>');
+        }
+        return value;
+      };
+
+      const rowKeys = [
```

Comment:
The whole function is way too complex. Consider creating helper functions for the constants which require complex conditions.
Consider these best practices for your refactor: 
- Try to avoid ternaries as they are harder to read/understand. 
- Try to avoid such cases `Object.entries(propsMeta ?? {}).map` since the map function will be called at least once when `propsMeta` is `undefined` or `null` which is useless.
- `if/elseif/elseif/else` could be avoided by using returns in if-conditions inside of a function
- Move typings out of the function to the top of the file eg. `<
        T extends { isDeprecated?: boolean; isBreakpointCustomizable?: boolean; isExperimental?: boolean },
      >(`

Remember that other colleagues have to understand it so it's your challenge to support them by writing clean and structured code which is easily understandable. You already did a great job reducing the complexity after rewriting but there is more potential.